blockprotocol/blockprotocol
# Implement `/account/general` page

## 🌟 What is the purpose of this PR?


## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204086167143889/1204065050321097/f) _(internal)_
- [Asana task for drag-drop upload](https://app.asana.com/0/1204086167143892/1204323465641761/f) _(internal)_
- [Asana task for setting default font-family](https://app.asana.com/0/1204086167143892/1204300160585817/f) _(internal)_
- [Design](https://www.figma.com/file/Fpmde3E9nJMb4VLkPTYgrM/blockprotocol.org?node-id=4859%3A37452&t=YBUEWkEmsPE543E7-1) _(internal)_





## 🔍 What does this change?

- Redirect `/account` -> `/account/general`
- Create components for account panel views & general panel (e.g. `PanelSection`, `AvatarButton`, `AvatarWithOverlay` ...)
- Remove duplicated `Avatar` component, use `UserAvatar` instead
- Create `SaveOnBlurTextField` component
- Update MUI theme options around `TextField` to match the designs
- Set default font family as `Inset` on MUI theme.
- Create api routes
  - `GET remove-user-avatar`
  - `POST update-user-preferred-name`
- Use `/general` as fallback if no panel slug is deteched by opening `/account` URL via client-side routing (not happening on codebase for now, but just in case), 
- Implement panel selector dropdown for mobile at `/account`


## Next Step

- [Implement drag-drop avatar upload](https://app.asana.com/0/1204086167143892/1204323465641761/f) _(internal)_

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Open `/account/general` page
1.  Confirm that it looks like design & functions property

## 📹 Demo

https://user-images.githubusercontent.com/39553853/229517641-8f4d0d70-b909-4b2f-9141-60de367d7497.mov

![Screenshot 2023-04-03 at 16 02 11](https://user-images.githubusercontent.com/39553853/229517680-ba845a86-3952-433b-845e-b0a381edcfa2.png)

https://user-images.githubusercontent.com/39553853/229517686-0acdee2e-d555-41e6-a519-f72656514932.mov

https://user-images.githubusercontent.com/39553853/229715865-20072977-2be6-48fb-9235-42c3e232e9c4.mov
